libminifi/include/utils/OptionalUtils.h

```diff
@@ -0,0 +1,46 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LIBMINIFI_INCLUDE_UTILS_OPTIONALUTILS_H_
+#define LIBMINIFI_INCLUDE_UTILS_OPTIONALUTILS_H_
+
+#include <utility>
+
+#include <nonstd/optional.hpp>
+#include "utils/gsl.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+using nonstd::optional;
+using nonstd::nullopt;
+
+template<typename T>
+optional<typename gsl_lite::remove_cvref<T>::type> optional_from_nullable(T&& obj) {
+  return obj ? optional<typename gsl_lite::remove_cvref<T>::type>{ std::forward<T>(obj) } : nullopt;
```

Comment:
This should be a comparison against `nullptr`. Normally I prefer pointer-like objects to be checked with `bool` conversion, but this template is too generic and I wouldn't want an empty `optional` on any object that's convertible to `bool` and takes the value `false`.

Reply:
Updated the function to work on pointer types only.